### PR TITLE
feat(Text Classifier Node): Add Text Classifier Node

### PR DESCRIPTION
**Pull request by jeanpaul**
## Summary

This change adds the Text Classifier Node, that allows the user to define categories into which the input must be classified. Outputs are defined for each category + optional fallback output. The main system prompt can be changed, as well as whether or not each item can have multiple labels.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/AI-212/add-a-classification-node

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+								default: '',
+								description: 'Category to add',
+							},
+							{
+								displayName: 'Description',
+								name: 'description',
+								type: 'string',
+								default: '',
+								description: "Describe your category if it's not obvious",
+							},
```
Comment (OlegIvaniv) [inline packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts:95]: Neither of these fields is required; is that intentional? IMO at least `category` should be mandatory

File: packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+		const llm = (await this.getInputConnectionData(
+			NodeConnectionType.AiLanguageModel,
+			0,
+		)) as BaseLanguageModel;
+
+		const categories = (
+			this.getNodeParameter('categories', 0) as {
+				categories: [{ category: string; description: string }];
+			}
+		)?.categories;
```
Comment (OlegIvaniv) [inline packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts:184]:
```suggestion
const categories = this.getNodeParameter('categories.categories', 0) as Array<{
			category: string;
			description: string;
		}>;
```

**Review by OlegIvaniv — COMMENTED** [review]
Nice one, works like a charm! Left a few comments to discuss

File: packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+				type: 'boolean',
+				default: false,
+			},
+			{
+				displayName: 'Add Fallback Option',
+				name: 'fallback',
+				type: 'boolean',
+				default: false,
+				description: 'Whether to add a "fallback" option if no other categories match',
+			},
```
[the remark this answers is not in the document]
Comment (jeanpaul) [reply packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts:112]: Sure, good idea. I'll get on that!

File: packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+					{
+						// eslint-disable-next-line n8n-nodes-base/node-param-display-name-miscased
+						name: 'Define below',
+						value: 'define',
+						description:
+							'Use an expression to reference data in previous nodes or enter static text',
+					},
+				],
+				default: 'auto',
+			},
```
[the remark this answers is not in the document]
Comment (jeanpaul) [reply packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts:133]: Good point. I didn't really know what default to pick; but it makes sense to use a text parameter indeed.

File: packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts (hunk 2 above)
Comment (jeanpaul) [reply packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts:184]: ah, you can request sub-parameters, thx for that!

File: packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeanpaul) [reply packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts]: So I changed this, put in a default, not sure if that's common... Can you check if this behaves as expected for you?

[the remark this answers is not in the document]
Comment (OlegIvaniv) [reply packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts]: I would keep it empty, just like any other text parameter. `={{ $json.text }}` expression makes assumptions about the previous node, which will not be correct in most cases(as we don't know if the input is really in the `text` property). Btw, maybe `Text to Classify` or just `Text` would be a better Display Name for this field. As `Input Prompt` sounds like you can somehow control the behavior of the LLM, but in this case, it's just the input text for classification. Other than it still works as expected

**Review by OlegIvaniv — DISMISSED** [review]
Thanks for addressing all my points! LGTM

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.51.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.51.0)

**Pull request merged**